docs/api.md

```diff
@@ -554,6 +564,187 @@ Response:
 }
 ```
 
+### `GET /api/saved_pages`
+
+Returns all a user's saved pages
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_pages`
+  - an array of pages the user has saved, identified by their page path
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.get_saved_pages(
+    govuk_account_session: "session-identifier",
+)
+```
+
+Response when no pages are saved:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": []
+}
+```
+
+Response when a user has saved two pages:
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_pages": [
+      { "page_path": "/government/organisations/government-digital-service" },
+      { "page_path": "/government/organisations/cabinet-office" },
+  ]
+}
+```
+
+### `POST /api/saved_pages`
+
+Create a saved page in a user's account
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### Request parameters
+
+- `page_path`
+  - An escaped URL safe string that identifies the path of a saved page.
+
+#### JSON response fields
+
+- `govuk_account_session` *(optional)*
+  - a new session identifier
+- `saved_page`
+  - an object containing the page path of the successfully saved page
+
+#### Response codes
+
+- 401 if the session identifier is invalid
+- 422 if page path is nil or empty
+- 422 if page path is not an escaped URL safe string
+- 422 if page path is not just a path
+- 422 if the page could not otherwise be successfully saved
+- 200 otherwise
+
+#### Example request / response
+
+Request (with gds-api-adapters):
+
+```ruby
+GdsApi.saved_page_api.set_saved_page(
+    page_path: "/guidance/foo",
+    govuk_account_session: "session-identifier",
+)
+```
+
+```json
+{
+    "govuk_account_session": "YWNjZXNzLXRva2Vu.cmVmcmVzaC10b2tlbg==",
+    "saved_page": {
+      "page_path": "/guidance/foo"
+    },
+}
+```
+
+### `DELETE /api/saved_pages/:page_path`
+
+Remove a saved page from a user's account
+
+#### Request headers
+
+- `GOVUK-Account-Session`
+  - the user's session identifier
+
+#### Request parameters
+
+- `page_path`
+  - An escaped URL safe string that identifies the path of a saved page.
```

Comment:
This is also a path param

Reply:
Hmm, I guess that's implicit from the URL in the section title actually... so this is fine